cmake/CMakeLists.txt

```diff
@@ -1029,8 +1043,11 @@ if (WIN32)
 else()
   add_definitions(-DPLATFORM_POSIX)
   # Enable warning and enable space optimization in Linux
-  string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -ffunction-sections -fdata-sections")
-  string(APPEND CMAKE_C_FLAGS " -Wall -Wextra -ffunction-sections -fdata-sections")
+  if (NOT onnxruntime_ENABLE_BITCODE)
+    # Cannot use -ffunction-sections -fdata-sections if we enable bitcode
+    string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -ffunction-sections -fdata-sections")
+    string(APPEND CMAKE_C_FLAGS " -Wall -Wextra -ffunction-sections -fdata-sections")
+  endif()
 
```

Comment:
Shouldn't we still set the warning flags?

Can we also just set -ffunction-sections and -fdata-sections in one place early on instead of here and in the minimal build setup?